src/main/scala/net/akmorrow13/endive/pipelines/DatasetCreationPipeline.scala

```diff
@@ -91,36 +93,46 @@ object DatasetCreationPipeline extends Serializable  {
     val windowSize = 200
     val stride = 50
 
-    val train: RDD[(String, String, ReferenceRegion, Int)] = Preprocess.loadLabelFolder(sc, labelsPath)
-        .cache()
+    val fs: FileSystem = FileSystem.get(new Configuration())
+    val labelStatus = fs.listStatus(new Path(labelsPath))
+    println(s"first label file: ${labelStatus.head.getPath.getName}")
+    val dnaseStatus = fs.listStatus(new Path(dnasePath))
+    println(s"first dnase file: ${dnaseStatus.head.getPath.getName}")
 
-    println("First reading labels")
-    train.count()
+      for (i <- labelStatus) {
+        val file: String = i.getPath.getName
+        try {
+          val (train: RDD[(String, String, ReferenceRegion, Int)], cellTypes: Array[String]) = Preprocess.loadLabels(sc, file)
+          val tf = train.first._1
+          println(s"celltypes for tf ${tf}, file ${file}:")
+          cellTypes.foreach(println)
 
-    // extract sequences from reference over training regions
-    val sequences: RDD[LabeledWindow] = extractSequencesAndLabels(referencePath, train).cache()
+          // extract sequences from reference over training regions
+          val sequences: RDD[LabeledWindow] = extractSequencesAndLabels(referencePath, train).cache()
 
-    // Load DNase data of (cell type, peak record)
-    val dnase: RDD[(String, PeakRecord)] = Preprocess.loadPeakFolder(sc, dnasePath)
-          .map(r => (Window.filterCellTypeName(r._1), r._2))
-      .cache()
+          // Load DNase data of (cell type, peak record)
+          val dnaseFiles = dnaseStatus.filter(r => {
+            cellTypes.contains(r.getPath.getName.split(".")(1))
+          })
+          val dnase: RDD[(String, PeakRecord)] = Preprocess.loadPeakFiles(sc, dnaseFiles.map(_.getPath.getName))
+            .map(r => (Window.filterCellTypeName(r._1), r._2))
+            .cache()
 
-//    // load rnase data
-//    val rnaLoader = new RNAseq(genes, sc)
-//    val rnaseq: RDD[(String, RNARecord)] = rnaLoader.loadRNAFolder(sc, rnaseqPath)
-//      .cache()
+          val sd = DatasetCreationPipeline.getSequenceDictionary(referencePath)
 
-    val sd = DatasetCreationPipeline.getSequenceDictionary(referencePath)
+          val cellTypeInfo = new CellTypeSpecific(windowSize, stride, dnase, sc.emptyRDD[(String, RNARecord)], sd)
 
-    val cellTypeInfo = new CellTypeSpecific(windowSize, stride, dnase, sc.emptyRDD[(String, RNARecord)], sd)
+          val fullMatrix: RDD[LabeledWindow] = cellTypeInfo.joinWithDNase(sequences)
 
-    val fullMatrix: RDD[LabeledWindow] = cellTypeInfo.joinWithDNase(sequences)
+          // save data
+          val output = aggregatedSequenceOutput + s"_${tf}"
```

Comment:
This is sort of confusing since the output directory thats created is decided by the TFs, I'd recommended saving the dataset in aggregatedSequenceOutput/${tf} or something all the lines so we have a stable API for 1 vs many TFs